File: pages/ultimate-learnapp-solver/solver.js
const cards = document.querySelector('iframe[src^="https://learningapps.org/show.php"]').contentDocument.querySelectorAll('.cardOuter .card');
let guid_list = [];
let regex = /openApp\('https:\/\/learningapps\.org\/watch\?v=([^']+)/
cards.forEach(card => {
    guid_list.push(card.getAttribute('onclick').match(regex)[1]);
});

const t = new URL(location.href).searchParams.get('v');
const time = 5;
const userName = document.querySelector('iframe[src^="https://learningapps.org/show.php"]').contentDocument.querySelector('#usernameDisplay').innerText;

const headers = {
    "accept": "*/*",
    "content-type": "application/x-www-form-urlencoded;charset=UTF-8",
    "sec-ch-ua": "\"Chromium\";v=\"130\", \"Microsoft Edge\";v=\"130\", \"Not?A_Brand\";v=\"99\"",
    "sec-ch-ua-mobile": "?0",
    "sec-ch-ua-platform": "\"Windows\"",
    "sec-fetch-dest": "empty",
    "sec-fetch-mode": "cors",
};

async function sleep(ms) {
    return new Promise((resolve) => setTimeout(resolve, ms));
}

async function processGuids(guid_list) {
    for (const guid of guid_list) {
        console.log(guid);
        
        await fetch("https://learningapps.org/setAppSolved.php", {
            method: "POST",
            headers: headers,
            body: `guid=${guid}&value=100&time=${time}&user=0`
        })
        .then(response => response.json())
        .then(data => console.log(data))
        .catch(error => console.error('Error:', error));
        
        await fetch("https://learningapps.org/collection.php", {
            method: "POST",
            headers: headers,
            body: `u=${userName}&c=${t}`
        })
        .then(response => response.json())
        .then(data => console.log(data))
        .catch(error => console.error('Error:', error));
        
        await fetch("https://learningapps.org/collection.php", {
            method: "POST",
            headers: headers,
            body: `u=${userName}&a=${guid}&c=${t}&t=${time}`
        })
        .then(response => response.json())
        .then(data => console.log(data))
        .catch(error => console.error('Error:', error));
    }
}

processGuids(guid_list);

File: pages/ultimate-learnapp-solver/solver.py
import requests
from bs4 import BeautifulSoup

userName = 'as'
time = 5
guid = 'peeg458gc23'
t = 'p80pwfpyt23'
url = 'https://learningapps.org/show.php?id=prvfq8ft324'

headers = {
    "accept": "*/*",
    "accept-language": "de,en-US;q=0.9,en;q=0.8,zh-CN;q=0.7,zh;q=0.6",
    "content-type": "application/x-www-form-urlencoded;charset=UTF-8",
    "sec-ch-ua": "\"Chromium\";v=\"130\", \"Microsoft Edge\";v=\"130\", \"Not?A_Brand\";v=\"99\"",
    "sec-ch-ua-mobile": "?0",
    "sec-ch-ua-platform": "\"Windows\"",
    "sec-fetch-dest": "empty",
    "sec-fetch-mode": "no-cors",
}

# First fetch request
response1 = requests.post(
    "https://learningapps.org/setAppSolved.php",
    headers=headers,
    data=f"guid={guid}&value=100&time={time}&user=0"
)

# Second fetch request
response2 = requests.post(
    "https://learningapps.org/collection.php",
    headers=headers,
    data=f"u={userName}&c={t}"
)

# Third fetch request
response3 = requests.post(
    "https://learningapps.org/collection.php",
    headers=headers,
    data=f"u={userName}&a={guid}&c={t}&t={time}"
)

# print(response1.status_code, response2.status_code, response3.status_code)

redirect_window = requests.get(url, headers=headers).text
print(redirect_window)
# redirect_window_obj = BeautifulSoup(redirect_window, 'html.parser')
# redirect_window_obj.select('script[src^="https://learningapps.org/data"]').get('src')

